### PR TITLE
More flexible Video Texture

### PR DESCRIPTION
**Pull request by edzis**
Allows to override `autoPlay` and `loop` values
Allows to use manual texture updating
Maintains backwards-compatibility

As a fallow-up on http://www.html5gamedevs.com/topic/34582-videotexture-1-frame-per-render-loop/?do=findComment&comment=201701

### CODE REVIEW

**Comment by edzis** [comment]
@deltakosh I would have removed `name` from constructor and used just use the computed value from initial `src`, but that would be breaking backward's compatibility.

For advanced usage see

    this._scene.onBeforeRenderObservable.add(this.updateVideoTextures)

    private updateVideoTextures = (): void => {
      const activeMeshes = this._scene.getActiveMeshes()
      each(activeMeshes, mesh => {
          const material = mesh.material as StandardMaterial
          const texture = material.diffuseTexture
          if (texture instanceof VideoTexture) {
              texture.updateTexture(true)
          }
      })
    }

File: src/Materials/Textures/babylon.videoTexture.ts [hunk 1]
```diff
@@ -1,44 +1,106 @@
...
+                autoPlay: true,
+                loop: true,
+                autoUpdateTexture: true,
+            }
+        ) {
             super(null, scene, !generateMipMaps, invertY);
 
-            var urls: Nullable<string[]> = null;
-            this.name = name;
+            this._engine = (<Scene>this.getScene()).getEngine();
```
Comment (deltakosh) [inline src/Materials/Textures/babylon.videoTexture.ts]: You can use this.getScene()!.

**Comment by edzis** [comment]
@deltakosh The latest CI error make no sense to me.

File: src/Materials/Textures/babylon.videoTexture.ts [hunk 2]
```diff
@@ -1,44 +1,116 @@
...
+                source.src = url;
+                video.appendChild(source);
+            });
+        }
+        return video;
+    };
+
     export class VideoTexture extends Texture {
-        public video: HTMLVideoElement;
+        readonly autoUpdateTexture: boolean;
```
Comment (deltakosh) [inline src/Materials/Textures/babylon.videoTexture.ts]: this is public by default and thus it requires a comment (and please add public keyword)

**Comment by edzis** [comment]
@deltakosh I added comments on those 2 properties, but I don't get what is the issue behind errors about inner props of Settings.

**Comment by deltakosh** [comment]
Oh yeah you have to comment on a per property basis instead of using the @property tag

**Comment by edzis** [comment]
Finally...
Thanks for the tip!

**Comment by deltakosh** [comment]
Please update the what's new to credit yourself :)

**Pull request merged**